state/services.go

```diff
@@ -21,15 +21,15 @@ type BlockStore interface {
 	LoadBlockMeta(height int64) *types.BlockMeta
 	LoadBlock(height int64) *types.Block
 
-	SaveBlock(block *types.Block, blockParts *types.PartSet, seenCommit *types.Commit)
+	SaveBlock(block *types.Block, blockParts *types.PartSet, seenCommit types.SeenCommit)
 
 	PruneBlocks(height int64) (uint64, error)
 
 	LoadBlockByHash(hash []byte) *types.Block
 	LoadBlockPart(height int64, index int) *types.Part
 
-	LoadBlockCommit(height int64) *types.Commit
-	LoadSeenCommit(height int64) *types.Commit
+	LoadBlockCommit(height int64) *types.BlockCommit
+	LoadSeenCommit(height int64) types.SeenCommit
```

Comment:
why is it switching from pointer to non pointer?

Reply:
Ah, because `types.SeenCommit` is an interface.